src/TConsole.cpp

```diff
@@ -650,6 +653,69 @@ void TConsole::Command_Status(const std::string&, const std::vector<std::string>
     Application::Console().WriteRaw(Status.str());
 }
 
+void TConsole::Autocomplete_Lua(const std::string& stub, std::vector<std::string>& suggestions) {
+    auto stateNames = mLuaEngine->GetLuaStateNames();
+
+    for (const auto& name : stateNames) {
+        if (name.find(stub) == 0) {
+            suggestions.push_back("lua " + name);
+        }
+    }
+}
+
+void TConsole::Autocomplete_Kick(const std::string& stub, std::vector<std::string>& suggestions) {
+    std::string stub_lower = boost::algorithm::to_lower_copy(stub);
+
+    auto LowercaseCompare = [](std::string Name1, std::string Name2) -> bool {
+        std::string NameLower = boost::algorithm::to_lower_copy(Name1);
+        return StringStartsWith(NameLower, Name2);
+    };
+
+    mLuaEngine->Server().ForEachClient([&](std::weak_ptr<TClient> Client) -> bool {
+        auto Locked = Client.lock();
+        if (Locked) {
+            if (LowercaseCompare(Locked->GetName(), stub_lower)) {
+                suggestions.push_back("kick " + Locked->GetName());
+            }
+        }
+        return true;
+    });
+
+}
+
+void TConsole::Autocomplete_Settings(const std::string& stub, std::vector<std::string>& suggestions) {
+    const std::string subcommands[] = { "help", "list", "set", "get" };
+
+    auto [command, args] = ParseCommand(stub);
+
+    std::string arg;
+    if (!args.empty()) arg = boost::algorithm::to_lower_copy(args.at(0));
+
+    auto LowercaseCompare = [](std::string Name1, std::string Name2) -> bool {
```

Comment:
See comment above - either this can be a free function here somewhere, or it needs `const&`